microservices/first/main.py

```diff
@@ -0,0 +1,78 @@
+from flask import Flask, request, jsonify
+from utils import DBManager
+import json
+
+from microservices.first import utils
+
+DATABASE_FILE: str = 'deployments.db'
+app = Flask(__name__)
+db = DBManager(DATABASE_FILE)
+
+@app.before_first_request
+def initialize():
+    """Initialize the database and tables."""
+    db.create_deployments_table()
+    db.create_results_table()
+
+@app.route('/start', methods=['POST'])
+def start_deployment():
+    """The `start` endpoint. Takes in the url and branch of the repository to 
+    build and execute. Spawns a new leader in the form of a Docker container 
+    and execute the SPEED deployment. Returns the id of the newly created 
+    worker."""
+    data = request.json
+    deployment_id = db.add_deployment(data['url'], data['branch'])
+
+    url = request.form['url']
+    branch = request.form['branch']
+    worker_id = utils.run_docker_container(url, branch, 2, "ghcr.io/jakejack13/speed-leaders:latest")
+
```

Comment:
Worker ID needs to either be kept in the deployments database table or needs to be the deployment ID as a primary key.

Reply:
done

Reply:
Good point, completely my fault. Feel free to change it to `leader_id`. As for the return type, Docker container ids are hexadecimal strings. I'm sure you can convert back and forth between string and int though

Reply:
We should make it leader_id that's true I will make the change. It's a string because the ID is generated based on the docker container ID (or rather the first 12 characters), which docker generates. We can change this potentially, but we want to keep track of the docker container.